firebase.agent.nut

```diff
@@ -384,51 +384,95 @@ class Firebase {
     }
 
     // parses event messages
-    function _parseEventMessage(text) {
+    // (https://www.w3.org/TR/eventsource/#parsing-an-event-stream)
+    // Message example:
+    // event: put
+    // data: {"path": "/c", "data": {"foo": true, "bar": false}}
+    // All messages except errors have two lines
+    // function can parse several messages, not full message or both
+    function _parseEventMessage(input) {
+        local text = _bufferedInput + input;
+        _bufferedInput = "";
+        server.log(text);
         // split message into parts
-        local alllines = split(text, "\n");
-        if (alllines.len() < 2) return [];
-        local returns = [];
-        for (local i = 0; i < alllines.len(); ) {
-            local lines = [];
+        local allLines = split(text, "\n");
 
-            lines.push(alllines[i++]);
-            lines.push(alllines[i++]);
-            if (i < alllines.len() && alllines[i+1] == "}") {
-                lines.push(alllines[i++]);
+        // Check, if we have at least one message
+        if (allLines.len() < 2) {
+            _bufferedInput = text;
+            return [];
+        }
+
+        local parsedEvents = [];
+
+        for (local i = 0; i < allLines.len(); ) {
+            local lines = [];
+            //try to get 2 lines
+            if (i + 1 < allLines.len()) {
+                lines.push(allLines[i++]);
+                lines.push(allLines[i++]);
+            } else {
+                // check, if we have at least one line, that we should to save
+                if (i < allLines.len()) {
+                    isEndLine = text[text.len() - 1] == "\n";
+                    _bufferedInput = lines[i] + (isEndLine ? "\n" : "");
```

Comment:
No, we operate with only one line (and it is the last line) in this case, and we want to add "\n" only if last text character contains "\n".